internal/batches/watchdog/watchdog.go

```diff
@@ -0,0 +1,39 @@
+package watchdog
+
+import (
+	"time"
+)
+
+type WatchDog struct {
+	interval time.Duration
+	ticker   *time.Ticker
+	done     chan struct{}
```

Comment:
Could this be simplified to a single `*time.Ticker`, with `Stop` calling `w.ticker.Stop()` and `Start()` having something like:

```go
for range <-w.ticker.C { go w.callback() }
```

(This isn't a rhetorical question; I've mostly been thinking about Rust channels recently, and they have different semantics.)

Reply:
Oh nice! Didn't know one could do that.
Admittedly, (I shared with @Piszmog earlier), channels are one part of Go I'm still not very familiar with but this ticket helped me gain a bit of understanding there.